### PR TITLE
KEYCLOAK-5830 Automated stress test

### PR DESCRIPTION
**Pull request by tkyjovsk**
Added a `stress-test-loop.sh` script.

### CODE REVIEW

File: testsuite/performance/stress-test-loop.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+if [ -z $RUN_USERS_INITIAL ]; then RUN_USERS_INITIAL=60; fi
```
Comment (hmlnarik) [inline testsuite/performance/stress-test-loop.sh]: Can be replaced with
```bash
RUN_USERS_INITIAL=${RUN_USERS_INITIAL:-60}
```

File: testsuite/performance/stress-test-loop.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    echo "============"
+    echo
+    echo "   ITERATION:  $((i+1)) / $MAX_ITERATIONS"
+    echo "   USERS:      $RUN_USERS"
+    echo
+    echo "   $TEST_COMMAND"
+    echo
+    echo
+
+    eval "$TEST_COMMAND"
```
Comment (hmlnarik) [inline testsuite/performance/stress-test-loop.sh]: Lines 24-26 can be replaced with
```bash
if ! $TEST_COMMAND; then
```

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Please remove so many empty lines in the beginning and end of every output.

Furthermore, this seems like having O(n) complexity with most likely no issue found in the first rounds. Have you considered O(log N) algorithm (for example bisection) with initial parameters considered safely within and beyond the actual performance, and in e.g. 10 rounds getting closer and closer to the actual maximal number of users?

**Comment by tkyjovsk** [comment]
O(log n) would be nice to have but I'm not sure if it is a good idea to put the system under such extreme load spikes which would result from first rounds of the bisection algorithm. This way the load is increased at least somewhat gradually which reduces the risk of "breaking" something in the server.

**Comment by hmlnarik** [comment]
I don't understand the last comment. Stress test is exactly about finding out the boundary. Should we be afraid to reach it?

**Comment by tkyjovsk** [comment]
If we overstress the server too much in the initial iteration it may break something and subsequent iteration which would pass under normal circumstances, could fail. Leading to incorrect result.

**Comment by hmlnarik** [comment]
You need to rerun the *whole* test in clean installation for every new parameterization anyway, otherwise the previous runs could interfere, leading to incorrect results.

**Comment by stianst** [comment]
@tkyjovsk @hmlnarik bump

**Comment by tkyjovsk** [comment]
Closing as outdated. Will create a new PR shorthly.

**Pull request closed without merging**